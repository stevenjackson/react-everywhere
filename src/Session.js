import React, { Component } from 'react';
//const Speaker stuff
class Speaker extends Component {
  render() {
    return <div>
      <div>{this.props.FirstName} {this.props.LastName}</div>
    </div>
  }
}
class Session extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showDetails: false
    }

    this.toggleState = this.toggleState.bind(this)
  }
  toggleState(e) {
    e.preventDefault();
    this.setState({
      showDetails: !this.state.showDetails
    })
  }
  abstractLink() {
    return <a href="#" onClick={this.toggleState}>{ this.state.showDetails ? "X" : "show abstract"}</a>
  }
  render() {
    const { Title, Abstract } = this.props
    return <div className="session">
      <div>{Title}</div>
      <Speaker { ...this.props.Speakers[0] }/>
      { this.state.showDetails && <div className="abstract">{Abstract}</div> }
      { this.abstractLink() }
    </div>
  }
}

Session.propTypes = {
  Title: React.PropTypes.string.isRequired,
  Abstract: React.PropTypes.string.isRequired,
  Speakers: React.PropTypes.array.isRequired,
}

export default Session
