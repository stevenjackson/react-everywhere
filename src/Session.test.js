import React from 'react';
import ReactDOM from 'react-dom';
import Session from './Session';
import renderer from 'react-test-renderer'

it('matches snapshot', () => {
  const output = renderer.create(<Session title="title" abstract="abstract"/>);
  expect(output).toMatchSnapshot()
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  const output = ReactDOM.render(<Session title="title" abstract="abstract"/>, div);
});
