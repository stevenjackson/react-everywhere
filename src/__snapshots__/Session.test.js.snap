exports[`test matches snapshot 1`] = `
<div>
  <div>
    title
  </div>
  <div>
    abstract
  </div>
</div>
`;
